src/components/AppComponents/AppHeaderBurgerMenuSections/AppHeaderBurgerMenuSections.test.tsx

```diff
@@ -29,24 +38,50 @@ describe("AppHeaderBurgerburgerMenuSections", () => {
     const list = within(firstSection!).getByRole("list");
     expect(list).toBeInTheDocument();
   });
-  it("does not render a sign out button when user is not logged in", () => {
-    renderWithProviders()(
-      <AppHeaderBurgerMenuSections burgerMenuSections={burgerMenuSections} />,
-    );
 
-    const signOutButton = screen.queryByRole("link", { name: "Sign out" });
-    expect(signOutButton).not.toBeInTheDocument();
-  });
-  it("renders a sign out button when a user is logged in", async () => {
-    renderWithProviders({
-      ...allProviders,
-      user: {
-        user: { name: "user" },
-      },
-    })(<AppHeaderBurgerMenuSections burgerMenuSections={burgerMenuSections} />);
-    const signOutButton = await screen.findByRole("link", {
-      name: "Sign out",
+  describe("when the `use-auth-owa` feature is enabled", () => {
+    function SignOutButton({ children }: PropsWithChildren) {
+      return <>{children}</>;
+    }
+    SignOutButton.displayName = "SignoutButton";
+
+    beforeEach(() => {
+      jest.spyOn(posthog, "useFeatureFlagEnabled").mockReturnValue(true);
+    });
+
+    it("does not render a sign out button when user is not logged in", () => {
+      jest
+        .spyOn(featureFlaggedClerk, "useFeatureFlaggedClerk")
+        .mockReturnValue({
+          ...featureFlaggedClerk.fakeClerkApi,
+          SignOutButton,
+          SignedIn: () => null,
+        });
```

Comment:
The setup for these tests is horrendous. Once the feature flag is gone this can be simplified to use a plain mock of the Clerk package.